controller/api/destination/traffic_split_adaptor.go

```diff
@@ -52,6 +52,8 @@ func (tsa *trafficSplitAdaptor) publish() {
 	merged := sp.ServiceProfile{}
 	if tsa.profile != nil {
 		merged = *tsa.profile
+	} else {
+		merged.Name = fmt.Sprintf("%s.%s.svc.%s", tsa.id.Name, tsa.id.Namespace, tsa.clusterDomain)
```

Comment:
If we're creating an empty service profile here that will be populated with only destination overrides, we set the name based off the `trafficSplitAdaptor`.

It's set here instead of `profileTranslator` because I think it makes more sense since we are already setting one field of the service profile (destination overrides). This way, we don't need to check for empty service profile names later.